File: backend/src/main/java/com/insilenceclone/backend/domain/product/entity/Product.java
package com.insilenceclone.backend.domain.product.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;


@Entity
@Getter
@Table(name = "products")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "product_id")
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false)
    private Integer price;

    @Column(nullable = false, length = 50)
    private String category;

    @Column(nullable = false, length = 255)
    private String imageUrl;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Builder
    public Product(String name, Integer price, String category, String imageUrl, String description) {
        this.name = name;
        this.price = price;
        this.category = category;
        this.imageUrl = imageUrl;
        this.description = description;
    }
}

File: frontend/src/api/product.js
import http from './http.js';

/**
 * 상품 상세 조회
 * GET /api/v1/product-details/{productId}
 */
export const fetchProductDetail = (productId) => {
    return http.get(`/product-details/${productId}`);
};

/**
 * 상품 목록 조회
 * GET /api/v1/products
 */
export const fetchProducts = (category) => {
    const params = category ? { category } : {};
    return http.get('/products', { params });
};
